opsgenie/resource_opsgenie_incident_template_test.go

```diff
@@ -0,0 +1,167 @@
+				log.Printf("Destroying incident template %s", value.Name)
+				deleteRequest := incident.DeleteIncidentTemplateRequest{IncidentTemplateId: value.IncidentTemplateId}
+				if _, err := client.DeleteIncidentTemplate(context.Background(), &deleteRequest); err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func TestAccOpsGenieIncidentTemplate_basic(t *testing.T) {
+	config := testAccOpsGenieIncidentTemplate_basic(randomName)
+	resource.Test(t, resource.TestCase{
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckOpsGenieIncidentTemplateDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckOpsGenieIncidentTemplateExists(),
+				),
+			},
+		},
+	})
+}
+
+func testCheckOpsGenieIncidentTemplateDestroy(s *terraform.State) error {
+	client, err := incident.NewClient(testAccProvider.Meta().(*OpsgenieClient).client.Config)
+	if err != nil {
+		return err
+	}
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "opsgenie_incident_template" {
+			continue
+		}
+		result, err := client.GetIncidentTemplate(context.Background(), &incident.GetIncidentTemplateRequest{})
+		if err != nil {
+			if result != nil {
+				for _, value := range result.IncidentTemplates["incidentTemplates"] {
+					incidentName := "Incident Template Name-" + randomName
+					if strings.HasPrefix(value.Name, incidentName) {
+						x := err.(*ogClient.ApiError)
+						if x.StatusCode != 404 {
+							return errors.New(fmt.Sprintf("Incident template still exists: %s", x.Error()))
+						}
+					}
+				}
+			}
+		} else {
+			return err
+		}
+	}
+	return nil
+}
+
+func testCheckOpsGenieIncidentTemplateExists() resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		client, err := incident.NewClient(testAccProvider.Meta().(*OpsgenieClient).client.Config)
+		if err != nil {
+			return err
+		}
+		result, err := client.GetIncidentTemplate(context.Background(), &incident.GetIncidentTemplateRequest{})
+		if err != nil {
+			if result != nil {
+				for _, value := range result.IncidentTemplates["incidentTemplates"] {
+					incidentName := "Incident Template Name-" + randomName
+					if strings.HasPrefix(value.Name, incidentName) {
+						log.Printf("Incident template found.")
+					} else {
+						return fmt.Errorf("incident template does not exist (and it should)")
+					}
+				}
+			} else {
+				return fmt.Errorf("incident template does not exist (and it should)")
+			}
+		} else {
+			return err
+		}
+		return nil
+	}
+}
+
+func testAccOpsGenieIncidentTemplate_basic(randomName string) string {
+	return fmt.Sprintf(`
+resource "opsgenie_user" "test" {
+  username  = "gtest_1-%s@opsgenie.com"
+  full_name = "Acceptance Test User"
+  role      = "User"
+}
+resource "opsgenie_team" "test" {
+  name        = "gtest_1-%s"
+  description = "This team deals with all the things"
+  member {
+    id        = opsgenie_user.test.id
+    role      = "admin"
+  }
+}
```

Comment:
We can create a team without user do we really need to create user as well ?

Reply:
yup. we can do that. team with no member should be fine.